src/lxc/attach.c

```diff
@@ -199,52 +201,60 @@ static int in_same_namespace(pid_t pid1, pid_t pid2, const char *ns)
 		if (errno == ENOENT)
 			return -EINVAL;
 
+		saved_errno = errno;
```

Comment:
This is not wrong, but all but one cases want to assign `errno`, so it would also be okay to move this down to after `out:`.  The caller would know whether `errno` makes sense from the return value anyway.

Reply:
@Blub Yes, it looks better. Thanks for your review.